apps/site/assets/ts/schedule/components/schedule-finder/FirstLastTimes.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactElement } from "react";
+import { SelectedDirection } from "../ScheduleFinder";
+import { HoursOfOperation } from "../__schedule";
+
+const getTimes = (
+  hoursOfOperation: HoursOfOperation,
+  serviceType: "week" | "saturday" | "sunday",
+  selectedDirection: SelectedDirection
+): [string, string] => [
+  hoursOfOperation[serviceType][selectedDirection!].first_departure,
+  hoursOfOperation[serviceType][selectedDirection!].last_departure
+];
+
+interface Props {
+  hoursOfOperation: HoursOfOperation;
+  selectedDirection: SelectedDirection;
+}
+
+const FirstLastTimes = ({
+  hoursOfOperation,
+  selectedDirection
+}: Props): ReactElement<HTMLElement> | null => {
+  if (selectedDirection === null) {
+    return null;
+  }
+
+  const timesList = [
+    [
+      "Monday - Saturday",
+      ...getTimes(hoursOfOperation, "week", selectedDirection)
```

Comment:
The types define a `"saturday"`, but we seem to ignore that here and assume the `"week"` times are also valid for Saturday — curious why that is.

Reply:
Oh I should have added a comment on that! We shall be enlightened in an imminent commit :)